ext/dom/php_dom.c

```diff
@@ -1302,6 +1302,10 @@ int dom_hierarchy(xmlNodePtr parent, xmlNodePtr child)
 {
 	xmlNodePtr nodep;
 
+	if (child->type == XML_DOCUMENT_NODE) {
```

Comment:
Nit: Indent seems to be off

Reply:
Ah, right, that's because the next if statement is indented with spaces instead of tabs.